src/contexts/auth-context.js

```diff
@@ -0,0 +1,186 @@
+      method: "post",
+      url: "/api/auth/login",
+      headers: { accept: "*/*" },
+      data: { email: formData.email, password: formData.password },
+    });
+  };
+
+  const loginAsGuest = () => {
+    operationLogin({
+      method: "post",
+      url: "/api/auth/login",
+      headers: { accept: "*/*" },
+      data: { email: "neog@cobratube.com", password: "neogcamp" },
+    });
+  };
+
+  useEffect(() => {
+    if (responseLogin !== undefined) {
+      setUserName(responseLogin.foundUser.firstName);
+      setIsUserLoggedIn(true);
+      setUserData({
+        firstName: responseLogin.foundUser.firstName,
+        lastName: responseLogin.foundUser.lastName,
+        email: responseLogin.foundUser.email,
+        password: formData.password || "cobrastore",
+      });
+      setFormData(initialFromState);
+      notifySuccess("Login Successful");
+      localStorage.setItem("cobraToken", responseLogin.encodedToken);
+      navigate(location.state?.from?.pathname || "/user");
+    }
+  }, [responseLogin]);
+  useEffect(
+    () => errorLogin !== "" && notifyError("Invalid Email or Password"),
+    [errorLogin]
+  );
+
+  //Signup
+  const {
+    response: responseSignup,
+    error: errorSignup,
+    operation: operationSignup,
+  } = useAxios();
+
+  const onSubmitSignup = (e) => {
+    e.preventDefault();
+    operationSignup({
+      method: "post",
+      url: "/api/auth/signup",
+      headers: { accept: "*/*" },
+      data: {
+        firstName: formData.firstName,
+        lastName: formData.lastName,
+        email: formData.email,
+        password: formData.password,
+      },
+    });
+  };
+
+  useEffect(() => {
+    if (responseSignup !== undefined) {
+      setUserName(
+        formData.firstName.charAt(0).toUpperCase() + formData.firstName.slice(1)
+      );
+      setIsUserLoggedIn(true);
+      setUserData({
+        firstName: responseSignup.createdUser.firstName,
+        lastName: responseSignup.createdUser.lastName,
+        email: responseSignup.createdUser.email,
+        password: responseSignup.createdUser.password,
+      });
+      setFormData(initialFromState);
+      notifySuccess("Signup Successful");
+      notifyInfo(
+        "CobraStore currently runs on mock backend so signup details won't persist on page reload"
+      );
+      localStorage.setItem("cobraToken", responseSignup.encodedToken);
+      navigate("/videos");
+    }
+  }, [responseSignup]);
+
+  useEffect(
+    () => errorSignup !== "" && notifyError("Email Already Exists"),
+    [errorSignup]
+  );
+
+  //First Page load verify
+  const { response: responseVerifyUser, operation: operationVerifyUser } =
+    useAxios();
+  useEffect(() => {
+    operationVerifyUser({
+      method: "post",
+      url: "/api/auth/verify",
+      headers: {
+        accept: "*/*",
+      },
+      data: { encodedToken: localStorage.getItem("cobraToken") },
+    });
+  }, []);
+  useEffect(() => {
+    if (responseVerifyUser !== undefined) {
```

Comment:
That won't be a good approach here. 
Let's say we make an API call and we get an empty array or object. 
So that will be treated as a falsy value and the `if` block won't run but I want it to run.

Reply:
Got it.